### PR TITLE
Add HiLive2 recipe.

### PR DESCRIPTION
**Pull request by tloka**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by tloka** [comment]
@bioconda/core please review and send me an invitation to join the team. Thanks!

File: recipes/hilive2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ number: 0
+ skip: True # [osx]
+
+requirements:
+ build:
+  - {{ compiler('cxx') }}
+  - cmake >=3.10
+ host:
+  - zlib
+  - boost =1.69.0
```
Comment (dpryan79) [inline recipes/hilive2/meta.yaml]: You'll want `boost-cpp` instead

File: recipes/hilive2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ name: "{{ name|lower }}"
+ version: "{{ version }}"
+
+source:
+ - folder: hilive2
+   url: https://gitlab.com/rki_bioinformatics/HiLive2/-/archive/v{{ version }}/HiLive2-v{{ version }}.tar.bz2
+   sha256: 47a8e0085ba0c19d672eecca7c5b70af8de55008fb4aec6dba24fa6a1c150137
+ - folder: seqan
+   url: https://github.com/seqan/seqan/archive/seqan-v2.3.2.tar.gz
+   sha256: 73ce4b344e6a259db1518547544c730f548bc0b32292b30e3a027146e158085e
```
Comment (dpryan79) [inline recipes/hilive2/meta.yaml]: Do you really need the source code? We package seqan in bioconda.

File: recipes/hilive2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  - {{ compiler('cxx') }}
+  - cmake >=3.10
+ host:
+  - zlib
+  - boost =1.69.0
+  - lz4
+ run:
+  - zlib
+  - boost =1.69.0
+  - lz4
```
Comment (dpryan79) [inline recipes/hilive2/meta.yaml]: Are you dynamically linking all of this? Cmake has a tendency to compile everything statically.

File: recipes/hilive2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+about:
+ home: https://gitlab.com/rki_bioinformatics/HiLive2
+ license: 3-clause BSD License
```
Comment (dpryan79) [inline recipes/hilive2/meta.yaml]: Is there by chance a copy of the license packaged with the source code?

**Comment by tloka** [comment]
Many thanks for your help @dpryan79 !

**Comment by dpryan79** [comment]
Thanks @tloka!

**Comment by dpryan79** [comment]
@bioconda-bot merge

**Pull request merged**